common/usb_cdc.c

```diff
@@ -44,20 +44,27 @@
 #define AT91C_EP_OUT_SIZE 0x40
 #define AT91C_EP_IN_SIZE  0x40
 
+// Language must always be 0.
+#define STR_LANGUAGE_CODES 0x00
+#define STR_MANUFACTURER   0x01
+#define STR_PRODUCT        0x02
+
 static const char devDescriptor[] = {
 	/* Device descriptor */
 	0x12,      // bLength
 	0x01,      // bDescriptorType
 	0x00,0x02, // Complies with USB Spec. Release (0200h = release 2.0)
 	0x02,      // bDeviceClass:    (Communication Device Class)
-	0x00,      // bDeviceSubclass: (unused at this time)
-	0x00,      // bDeviceProtocol: (unused at this time)
+	0x02,      // bDeviceSubclass: (Abstract Control Model)
+	           // Allows use on Win10+ without INF file. Reference:
+	           // https://msdn.microsoft.com/en-us/library/windows/hardware/dn707976%28v=vs.85%29.aspx

```

Comment:
I think the Windows site is misleading here. According to the USB Spec, the 0x02 is the correct setting for bInterfaceSubclass, not for bDeviceSubclass.

Reply:
Hm, wouldn't be the first time Microsoft's documentation is patently wrong, but it does explicitly mention "device" not "interface". I found (but misplaced) a reference that indicated that it was valid for both, but I'll have to dig that up again.